src/client/fedavg.py

```diff
@@ -82,7 +82,7 @@ def load_data_indices(self):
         self.valset.indices = self.data_indices[self.client_id]["val"]
         self.testset.indices = self.data_indices[self.client_id]["test"]
 
-    def train_with_eval(self):
+    def train_locally(self, evaluate_clients: bool = True):
```

Comment:
`train_locally` is not as clear as `train_with_eval`, I think. 🤔 Because we already has `train()` as the entrance of client training process.

If you think this method need renaming, rename it to be clearer.